test/e2e/openshift/tektonaddondeployment_test.go

```diff
@@ -0,0 +1,97 @@
+// +build e2e
+
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openshift
+
+import (
+	"testing"
+
+	"github.com/tektoncd/operator/test/utils"
+
+	"github.com/tektoncd/operator/test/client"
+
+	"github.com/tektoncd/operator/test/resources"
+)
+
+// TestTektonAddonsDeployment verifies the TektonAddons creation, deployment recreation, and TektonAddons deletion.
+func TestTektonAddonsDeployment(t *testing.T) {
+	clients := client.Setup(t)
+
+	crNames := utils.ResourceNames{
+		TektonPipeline:  "pipeline",
+		TektonTrigger:   "trigger",
+		TektonAddon:     "addon",
+		Namespace:       "",
+		TargetNamespace: "tekton-pipelines",
+	}
+
+	utils.CleanupOnInterrupt(func() { utils.TearDownPipeline(clients, crNames.TektonPipeline) })
+	defer utils.TearDownPipeline(clients, crNames.TektonPipeline)
```

Comment:
Can we add TearDown for triggers also as we are creating `triigers` here

Reply:
:heavy_check_mark: